data_model/src/transaction.rs

```diff
@@ -439,6 +439,42 @@ impl PartialOrd for TransactionValue {
     }
 }
 
+/// `TransactionExt` is used in `FindAllTransactions` query
+#[derive(Debug, Clone, PartialEq, Eq, Decode, Encode, Deserialize, Serialize, IntoSchema)]
+pub struct TransactionExt {
+    /// Transaction
+    pub tx_value: TransactionValue,
+    /// Hash of the block to which `tx` belongs to
+    pub block_hash: Hash,
+}
+
+impl TransactionExt {
+    #[inline]
+    /// Used to return payload of the transaction
+    pub fn payload(&self) -> &Payload {
+        self.tx_value.payload()
+    }
+}
+
+impl Ord for TransactionExt {
+    #[inline]
+    fn cmp(&self, other: &Self) -> Ordering {
+        self.payload()
+            .creation_time
+            .cmp(&other.payload().creation_time)
+    }
+}
+
+impl PartialOrd for TransactionExt {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
```

Comment:
Also `#[inline]`. This seems like a lot of syntax, but the actual assembly and MIR are quite small, so you win by inlining this comparison.